### DISCUSSION TITLE
importing the requests module

### DISCUSSION THREAD

**Discussion by visjble** [pyscript/pyscript#1163] [Q&A] [answered]
I tried the following:
```
    <py-script>
      import requests
      url = 'https://www.google.com'
      data = requests.get(url)
      print(data)
    </py-script>
```

with an error:
raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='www.google.com', port=443): Max retries exceeded with url: / (Caused by SSLError("Can't connect to HTTPS URL because the **SSL module is not available**."))

What is the best way to implement the requests?

**Comment by FabioRosado** [comment, marked as answer]
Hello @visjble  thank you for creating this discussion!

There are a few different ways you can do this. If you want to use `requests` you can install `pyodide-http` and patch requests. Here's an example code of what I mean:

```html
<py-config>
packages = ["requests",  "pyodide-http"]
</py-config>

<py-script>
import pyodide_http
pyodide_http.patch_all()

import requests
url = 'https://jsonplaceholder.typicode.com/posts/2'
data = requests.get(url)
print(data)
</py-script>
```

This is the easiest way to do it.  Alternatively, you can create your own requests module following the great guide created by @JeffersGlass - [How to make HTTP requests using `PyScript`, in pure Python](https://docs.pyscript.net/latest/guides/http-requests.html).

Please let us know if this solves your problem 😄

**Reply by FabioRosado** [reply]
Oh it might also be worth to mention that google will reject the XMLHttpRequest due to CORS if you are running locally. You may want to replace the URL with something like `https://jsonplaceholder.typicode.com/posts/2` for testing 😄

**Reply by visjble** [reply]
aahg, I thought that was going to be the case with CORS. What a pain in the Batoozy
